.github/workflows/flutter.yaml

```diff
@@ -3,37 +3,49 @@ on: [push, pull_request]
 
 jobs:
   code-check:
-    # Check if file exists, rather than repo name - as initial run fails before flutter create executed.
-    # https://github.community/t/is-the-hashfiles-function-actually-available-in-the-if-context/202545
-    if: ${{ always() && hashFiles('pubspec.yaml') }}
     runs-on: ubuntu-latest
 
     steps:
       - uses: actions/checkout@v1
-      - uses: subosito/flutter-action@v1
+          
+      - name: Check file existence
+        id: check_files
+        uses: andstor/file-existence-action@v1
         with:
-          channel: 'stable'
+          files: "pubspec.yaml"
 
+      - name: Setup flutter
+        if: steps.check_files.outputs.files_exists == 'true'
+        uses: subosito/flutter-action@v1
+        with:
+          channel: 'stable'
+      
       - name: Get dependencies
+        if: steps.check_files.outputs.files_exists == 'true'
```

Comment:
Perhaps we could try Github [Dependent Jobs](https://docs.github.com/en/actions/using-workflows/advanced-workflow-features#creating-dependent-jobs) to avoid adding the `if` to each job step?

Then we would have:
1. A job checking the presence of pubspec
2. Second job depending on the first one, which starts conditionally depending on the output of a previous job.

Docs on setting and using output from a prerequisite job: https://docs.github.com/en/actions/using-jobs/defining-outputs-for-jobs

Reply:
Thanks! The main thought here is that we'll probably have to checkout twice then? Wanted to avoid that situation and save resources.